src/main/java/com/example/music_player/storage/CloudStorageAmazonS3.java

```diff
@@ -86,13 +104,14 @@ public void delete(Source source) {
 
     @Override
     public boolean isExist(Source source) {
-        Boolean isObjectExist = false;
+        log.info("IN isExist() : method started");
         S3Object s3Object = s3Client.getObject(bucketName, source.getName());
```

Comment:
Did you verify if s3client has specific method for object existence check instead of fetching whole object?